applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp

```diff
@@ -0,0 +1,138 @@
+        
+        result_file_name += "_EigenResults";
+        
+        mpGidEigenIO = GidEigenIO::Pointer (new GidEigenIO( 
+                            result_file_name,
+                            GiD_PostBinary,
+                            MultiFileFlag::SingleFile,
+                            WriteDeformedMeshFlag::WriteUndeformed,
+                            WriteConditionsFlag::WriteConditions) );
+
+        KRATOS_ERROR_IF_NOT(mpGidEigenIO) << "EigenIO could not be initialized!" << std::endl;
+    }
+
+    void PostprocessEigenvaluesProcess::ExecuteBeforeSolutionLoop()
+    {
+        KRATOS_ERROR_IF_NOT(mpGidEigenIO) << " EigenIO is uninitialized!" << std::endl;
+
+        mpGidEigenIO->InitializeMesh(0.0);
+        mpGidEigenIO->WriteMesh(mrModelPart.GetMesh());
+        mpGidEigenIO->WriteNodeMesh(mrModelPart.GetMesh());
+        mpGidEigenIO->FinalizeMesh();
+    }
+
+    void PostprocessEigenvaluesProcess::ExecuteFinalize()
+    {
+        KRATOS_ERROR_IF_NOT(mpGidEigenIO) << " EigenIO is uninitialized!" << std::endl;
+
+        const auto& eigenvalue_vector = mrModelPart.GetProcessInfo()[EIGENVALUE_VECTOR];
+        const SizeType num_eigenvalues = eigenvalue_vector.size();
+
+        const SizeType num_animation_steps = mOutputParameters["animation_steps"].GetInt();
+        double angle = 0.0;
+        std::string label = "";
+
+        const SizeType num_requested_results = mOutputParameters["eigen_results"].size();
+        std::vector<std::string> requested_results(num_requested_results);
+
+        for (SizeType i=0; i<num_requested_results; ++i)
+            requested_results[i] = mOutputParameters["eigen_results"].GetArrayItem(i).GetString();
+
+        for (SizeType i=0; i < num_animation_steps; ++i)
+        {
+            angle = 2 * Globals::Pi * i / num_animation_steps;
+
+            for(SizeType j=0; j<num_eigenvalues; ++j)
+            {
+                label = GetLabel(eigenvalue_vector[j]);
+
+                for(auto& node : mrModelPart.Nodes())
+                {
+                    // Copy the eigenvector to the Solutionstepvariable. Credit to Michael Andre
+                    DofsContainerType& r_node_dofs = node.GetDofs();
+                    Matrix& r_node_eigenvectors = node.GetValue(EIGENVECTOR_MATRIX);
+
+                    KRATOS_ERROR_IF_NOT(r_node_dofs.size() == r_node_eigenvectors.size2()) 
+                        << "Number of results on node " << node.Id() << " is wrong" << std::endl;
+
+                    SizeType k = 0;
+                    for (auto& r_dof : r_node_dofs)
+                        r_dof.GetSolutionStepValue(0) = std::cos(angle) * r_node_eigenvectors(j,k++);
+                }
+                
+                // for (const auto& var_name : requested_results)
+                //     // I have to get the Variable form the Name somehow
+                //     mpGidEigenIO->WriteEigenResults(mrModelPart, var_name.GetVariable(), label, i);
```

Comment:
@KratosMultiphysics/technical-committee I would need the `KratosGlobals.GetVariable` function that is available in Pyhton